### PR TITLE
Fixes assertion on calling String builtin with symbol

### PR DESCRIPTION
**Pull request by legendecas**
Add implementation for ECMA-262 v6 21.1.1.1 section.

JerryScript-DCO-1.0-Signed-off-by: legendecas legendecas@gmail.com

### CODE REVIEW

File: jerry-core/ecma/builtin-objects/ecma-builtin-string.c [hunk 1]
```diff
@@ -22,6 +22,9 @@
 #include "ecma-helpers.h"
 #include "ecma-objects.h"
 #include "ecma-string-object.h"
+#ifndef CONFIG_DISABLE_ES2015_SYMBOL_BUILTIN
+#include "ecma-symbol-object.h"
+#endif
```
Comment (LaszloLango) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string.c]: missing comment: `#endif /* !CONFIG_DISABLE_ES2015_SYMBOL_BUILTIN */`

File: jerry-core/ecma/builtin-objects/ecma-builtin-string.c [hunk 2]
```diff
@@ -119,10 +125,19 @@ ecma_builtin_string_dispatch_call (const ecma_value_t *arguments_list_p, /**< ar
...
   {
     ret_value = ecma_make_magic_string_value (LIT_MAGIC_STRING__EMPTY);
   }
+#ifndef CONFIG_DISABLE_ES2015_SYMBOL_BUILTIN
+  /* 2.a */
+  else if (ecma_is_value_symbol(arguments_list_p[0]))
+  {
+    ret_value = ecma_get_symbol_descriptive_string (arguments_list_p[0]);
+  }
+#endif
```
Comment (LaszloLango) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string.c]: missing comment: `#endif /* !CONFIG_DISABLE_ES2015_SYMBOL_BUILTIN */`

File: jerry-core/ecma/builtin-objects/ecma-builtin-string.c (hunk 1 above)
Comment (legendecas) [reply jerry-core/ecma/builtin-objects/ecma-builtin-string.c]: Fixed :)

File: jerry-core/ecma/builtin-objects/ecma-builtin-string.c [hunk 3]
```diff
@@ -119,10 +125,19 @@ ecma_builtin_string_dispatch_call (const ecma_value_t *arguments_list_p, /**< ar
...
   ecma_value_t ret_value = ECMA_VALUE_EMPTY;
 
+  /* 1. */
   if (arguments_list_len == 0)
   {
     ret_value = ecma_make_magic_string_value (LIT_MAGIC_STRING__EMPTY);
   }
+#ifndef CONFIG_DISABLE_ES2015_SYMBOL_BUILTIN
+  /* 2.a */
+  else if (ecma_is_value_symbol(arguments_list_p[0]))
```
Comment (rerobika) [inline jerry-core/ecma/builtin-objects/ecma-builtin-string.c]: One space needed before the opening parenthesis.

Comment (legendecas) [reply jerry-core/ecma/builtin-objects/ecma-builtin-string.c]: Updated 🐒

**Pull request merged**